documentation-website/Writerside/topics/Deep-Dive-into-DAO.md

```diff
@@ -2,204 +2,9 @@
-<def title="CompositeIdTable">Multiple columns make up the table id</def>
-</deflist>
-
-To define a custom column type as the primary key and id, use a typed `IdTable` directly and override the `id` column:
-```kotlin
-object Directors : IdTable<String>("directors") {
-    override val id: Column<EntityID<String>> = varchar("id", 32).entityId()
-    val name = varchar("name", 50)
-
-    override val primaryKey = PrimaryKey(id)
-}
-
-class Director(id: EntityID<String>) : Entity<String>(id) {
-    companion object : EntityClass<String, Director>(Directors)
-    var name by Directors.name
-}
-```
-
-To define multiple columns as part of the primary key and id, use `CompositeIdTable` and mark each composite column using `entityId()`.
-Each component column will be available for CRUD operations either individually (as for any standard column) or all together as part of the `id` column:
-```kotlin
-object Directors : CompositeIdTable("directors") {
-    val name = varchar("name", 50).entityId()
-    val guildId = uuid("guild_id").autoGenerate().entityId()
-    val genre = enumeration<Genre>("genre")
-
-    override val primaryKey = PrimaryKey(name, guildId)
-}
-
-class Director(id: EntityID<CompositeID>) : CompositeEntity(id) {
-    companion object : CompositeEntityClass<Director>(Directors)
-    var genre by Directors.genre
-}
-```
-<include from="Table-Definition.topic" element-id="add-id-column-tip" />
-```kotlin
-object Guilds : UUIDTable("guilds")
-
-object Directors : CompositeIdTable("directors") {
-    val name = varchar("name", 50).entityId()
-    val guildId = reference("guild_id", Guilds)
-    val genre = enumeration<Genre>("genre")
-    
-    init {
-        addIdColumn(guildId)
-    }
-
-    override val primaryKey = PrimaryKey(name, guildId)
-}
-```
-
-## Basic CRUD operations
-### Create
-```kotlin
-val movie = StarWarsFilm.new {
-    name = "The Last Jedi"
-    sequelId = 8
-    director = "Rian Johnson"
-}
-```
-
-To provide a manual id value to a new entity, pass the value as an argument to the `id` parameter:
-```kotlin
-StarWarsFilm.new(id = 2) {
-    name = "The Rise of Skywalker"
-    sequelId = 9
-    director = "J.J. Abrams"
-}
-```
-If the entity is a `CompositeEntity`, the id value can be constructed by creating a component column-to-value association using `CompositeID`:
-```kotlin
-val newId = CompositeID {
-    it[Directors.name] = "J.J. Abrams"
-    it[Directors.guildId] = UUID.randomUUID()
-}
-
-Director.new(newId) {
-    genre = Genre.SCI_FI
-}
-```
-### Read
-To get entities use one of the following methods:
-```kotlin
-val movies = StarWarsFilm.all()
-val movies = StarWarsFilm.find { StarWarsFilms.sequelId eq 8 }
-val movie = StarWarsFilm.findById(5)
-```
-<tip>
-For a list of available predicates, see
-<a href="DSL-Querying-data.topic" anchor="where-expression">DSL Where expression</a>.
-</tip>
+The DAO (Data Access Object) API of Exposed, is similar to ORM frameworks like Hibernate with a Kotlin-specific API.
```

Comment:
Just curious about why this file hasn't been removed entirely since all(?) its content is now in other topics?

Also, it seems to be causing 2 broken link errors in: `About` and `Transactions`.

Reply:
My bad, it should have been removed.